### PR TITLE
Possible to have custom filenames for deployed files.

### PR DESCRIPTION
**Pull request by NilsOlavKJohansen**
Changelog: Now it is possible to to select a custom filename for the deployed files by using a FILE_NAME string in the config file.

Signed-off-by: nils olav <nils.olav@northern.tech>

# External Contributor Checklist

🚨 Please review the [guidelines for contributing](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md) to this repository.

- [ ] Make sure that all commits have a [`Changelog`](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md#changelog-tags) tag. If nothing should be added to the Changelog, add a `Changelog: None` tag. If there is a change, add `Changelog: Commit`, or `Changelog: Title`, depending on what should be included in the changelog.

- [ ] Make sure that all commits are signed with [`git --signoff`](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work). Also note that the signoff author must match the author of the commit.

### Description

Please describe your pull request.

Thank you!

### CODE REVIEW

File: configs/mender_convert_config [hunk 1]
```diff
@@ -78,6 +78,11 @@ MENDER_DATA_PART_GROWFS=y
 # input disk image
 MENDER_DEVICE_TYPE=""
 
+# A string that defines the filenames of the deployed files.
+# This is optional. If no FILE_NAME is defined the filenames will be a combination off
+# the date and the names of all DEVICE_TYPES. These will be seperated with a +.
+FILE_NAME=""
```
Comment (lluiscampos) [inline configs/mender_convert_config:84]: Technically, it is not used a a "file" name but as an "image" name, as mender-convert will be appending a `.img` and creating other related files like the Artifact or the tests configurations using this "image" name as a base name.

I suggest `IMAGE_BASE_NAME` or even better `DEPLOY_IMAGE_NAME` or `OUTPUT_IMAGE_NAME`.

File: mender-convert-package [hunk 2]
```diff
@@ -149,12 +149,18 @@ artifact_name=$(cat work/rootfs/etc/mender/artifact_info | sed 's/[^=]*=//')
 # Get the name from the input disk_image
 temp_img_name=$(basename "$disk_image")
 
+#Checks if user has chooses custom filename
```
Comment (lluiscampos) [inline mender-convert-package:152]: Nitpick
```suggestion
# Check if user has chooses custom filename
```

To keep comments style consistent

File: mender-convert-package [hunk 3]
```diff
@@ -149,12 +149,18 @@ artifact_name=$(cat work/rootfs/etc/mender/artifact_info | sed 's/[^=]*=//')
...
+if [ -z "${FILE_NAME}" ]; then
 # Strip the image suffix from temp_img_name and set the original image name as
 # the output image name
 image_name="${temp_img_name%.*}"
 
 #Adds device types to the filename and replaces whitespace with + to seperate device types.
```
Comment (lluiscampos) [inline mender-convert-package:158]: If you fix the previous nitpick, fix also here
```suggestion
# Add device types to the filename and replace whitespace with + to separate device types. 
```

**Review by lluiscampos — CHANGES_REQUESTED** [review]
I think the variable name needs to be renamed. See comment below. Remember to amend the `Changelog` in your commit message if we rename the variable.

File: configs/mender_convert_config (hunk 1 above)
Comment (NilsOlavKJohansen) [reply configs/mender_convert_config]: Thanks for good feed back Lluís, I appreciate it!

File: mender-convert-package [hunk 4]
```diff
@@ -149,12 +149,18 @@ artifact_name=$(cat work/rootfs/etc/mender/artifact_info | sed 's/[^=]*=//')
 # Get the name from the input disk_image
 temp_img_name=$(basename "$disk_image")
 
+# Check if user has chooses custom filename
+if [ -z "${DEPLOY_IMAGE_NAME}" ]; then
 # Strip the image suffix from temp_img_name and set the original image name as
 # the output image name
 image_name="${temp_img_name%.*}"
```
Comment (lluiscampos) [inline mender-convert-package:157]: Still missing indentation here, right?

```suggestion
# Check if user has chooses custom filename
if [ -z "${DEPLOY_IMAGE_NAME}" ]; then
  # Strip the image suffix from temp_img_name and set the original image name as
  # the output image name
  image_name="${temp_img_name%.*}"
  
```

**Review by lluiscampos — APPROVED** [review]
Looking good! :climbing: 

Still missing indentation inside the `if` statement, I think. Please double check. I am pre approving, though.

**Pull request merged**